library/src/main/java/com/antonyt/infiniteviewpager/InfiniteViewPager.java

```diff
@@ -39,7 +39,9 @@ public void setCurrentItem(int item, boolean smoothScroll) {
             super.setCurrentItem(item, smoothScroll);
             return;
         }
-        item = getOffsetAmount() + (item % getAdapter().getCount());
+        // offset only for the first time
+        if (item < getOffsetAmount())
```

Comment:
If the first action is to scroll back the offset is still added.
I would recommend to get half of offset amount
`if (item < getOffsetAmount() / 2)`

eg. when offset is 1000 and you scroll back - the function returns 1999 it would be better to return 999